### PR TITLE
Add reference to #comet channel in Arrow Rust Discord server

### PR DESCRIPTION
**Pull request by ajmarcus**
## Which issue does this PR close?

None (documentation only change)

## Rationale for this change

I wasn't sure where discussion about comet happens.

## What changes are included in this PR?

Documentation change to refer to the #comet channel in the Arrow Rust Discord server

## Are these changes tested?

No  - documentation change only

## Are there any user-facing changes?

Yes. Documentation change.

### CODE REVIEW

File: docs/source/contributor-guide/communication.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ meet other contributors and get guidance on where to contribute. It is important
 decisions are made fully in the open, on GitHub.
 
 Most of us use the `#datafusion` and `#arrow-rust` channels in the [ASF Slack
-workspace](https://s.apache.org/slack-invite) and the [Arrow Rust Discord
+workspace](https://s.apache.org/slack-invite) and the `#comet` channel in the [Arrow Rust Discord
 server](https://discord.gg/Qw5gKqHxUM) for discussions.
```
Comment (andygrove) [inline docs/source/contributor-guide/communication.md:42]: I think it is a good idea to make it easier to find subproject-specifc channels, so we should list all of them.

In Slack we have:

Screenshot from 2024-07-24 10-18-04

In Discord we have:

Screenshot from 2024-07-24 10-18-38

How about something like:

Most of us use the [ASF Slack workspace](https://s.apache.org/slack-invite) and the [Arrow Rust Discord
server](https://discord.gg/Qw5gKqHxUM) for discussions.

There are specific channels for Arrow, DataFusion, and the DataFusion subprojects (Ballista, Comet, Python).

Comment (ajmarcus) [reply docs/source/contributor-guide/communication.md:42]: Makes sense! Will update tonight.

Comment (ajmarcus) [reply docs/source/contributor-guide/communication.md:42]: updated the language and added all of the channels. think that is what you meant.

Comment (ajmarcus) [reply docs/source/contributor-guide/communication.md:42]: happy to change again if I misunderstood.

**Review by andygrove — APPROVED** [review]
Thanks @ajmarcus

**Comment by alamb** [comment]
🚀

**Pull request merged**